core/blockchain.go

```diff
@@ -470,9 +474,7 @@ func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, genesis *Genesis
 		}
 		rawdb.WriteChainConfig(db, genesisHash, chainConfig)
 	}
-	if bc.logger != nil && bc.logger.OnBlockchainInit != nil {
```

Comment:
Ah I remember now why I put it there, it's because the initialization is done and chain config has also been written to db at that point. But it does make sense to emit this before genesis block definitely.